widget/entry.go

```diff
@@ -536,6 +542,33 @@ func (e *Entry) TappedSecondary(pe *fyne.PointEvent) {
 	e.popUp.ShowAtPosition(popUpPos)
 }
 
+// TouchDown is called when this entry gets a touch down event on mobile device, we ensure we have focus.
+//
+// Since: 2.1
+//
+// Implements: mobile.Touchable
+func (e *Entry) TouchDown(*mobile.TouchEvent) {
+	if c := fyne.CurrentApp().Driver().CanvasForObject(e); c != nil {
```

Comment:
Although here has been used `super()` in c.Focus. I wonder if we need to use `super()` in `CanvasForObject()` too instead of just the entry. In my understanding when the entry widget is extending, the entry receiver as it does not exist in the canvas (only the new extended widget would exist), am I missing something? 🤔

Reply:
Yes you're right, fixed up.